src/doc/trpl/patterns.md

```diff
@@ -23,6 +23,31 @@ match x {
 
 This prints `one`.
 
+There’s one pitfall with patterns: like anything that introduces a new binding,
```

Comment:
I wonder if it's worth having a section on shadowing? It is interesting to point out that Rust typically allows shadowing, and patterns are just a special case.

It's ALSO interesting (perhaps) to point out that you cannot shadow a constant or enum variant name.